crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -536,9 +544,26 @@ impl ScheduleGraph {
                             densely_chained: true
                         }
                     };
+                    let more_than_one_entry = config_iter.len() > 0;
+                    let anonymous_set = if more_than_one_entry && !collective_conditions.is_empty()
```

Comment:
This doesn't handle the case where we have collective conditions in a single SystemConfigs::Configs (ex: `(foo, ).run_if(X)`

Reply:
Fixed in my commit below